parachains/runtimes/assets/statemine/src/xcm_config.rs

```diff
@@ -143,17 +149,122 @@ match_types! {
+				pallet_assets::Call::transfer_keep_alive { .. } |
+				pallet_assets::Call::force_transfer { .. } |
+				pallet_assets::Call::freeze { .. } |
+				pallet_assets::Call::thaw { .. } |
+				pallet_assets::Call::freeze_asset { .. } |
+				pallet_assets::Call::thaw_asset { .. } |
+				pallet_assets::Call::transfer_ownership { .. } |
+				pallet_assets::Call::set_team { .. } |
+				pallet_assets::Call::clear_metadata { .. } |
+				pallet_assets::Call::force_clear_metadata { .. } |
+				pallet_assets::Call::force_asset_status { .. } |
+				pallet_assets::Call::approve_transfer { .. } |
+				pallet_assets::Call::cancel_approval { .. } |
+				pallet_assets::Call::force_cancel_approval { .. } |
+				pallet_assets::Call::transfer_approved { .. } |
+				pallet_assets::Call::touch { .. } |
+				pallet_assets::Call::refund { .. },
+			) |
+			RuntimeCall::Uniques(
+				pallet_uniques::Call::create { .. } |
+				pallet_uniques::Call::force_create { .. } |
+				pallet_uniques::Call::destroy { .. } |
+				pallet_uniques::Call::mint { .. } |
+				pallet_uniques::Call::burn { .. } |
+				pallet_uniques::Call::transfer { .. } |
+				pallet_uniques::Call::freeze { .. } |
+				pallet_uniques::Call::thaw { .. } |
+				pallet_uniques::Call::freeze_collection { .. } |
+				pallet_uniques::Call::thaw_collection { .. } |
+				pallet_uniques::Call::transfer_ownership { .. } |
+				pallet_uniques::Call::set_team { .. } |
+				pallet_uniques::Call::approve_transfer { .. } |
+				pallet_uniques::Call::cancel_approval { .. } |
+				pallet_uniques::Call::force_item_status { .. } |
+				pallet_uniques::Call::set_attribute { .. } |
+				pallet_uniques::Call::clear_attribute { .. } |
+				pallet_uniques::Call::set_metadata { .. } |
+				pallet_uniques::Call::clear_metadata { .. } |
+				pallet_uniques::Call::set_collection_metadata { .. } |
+				pallet_uniques::Call::clear_collection_metadata { .. } |
+				pallet_uniques::Call::set_accept_ownership { .. } |
+				pallet_uniques::Call::set_collection_max_supply { .. } |
+				pallet_uniques::Call::set_price { .. } |
+				pallet_uniques::Call::buy_item { .. },
+			) => true,
+			_ => false,
+		}
+	}
+}
+
 pub type Barrier = DenyThenTry<
 	DenyReserveTransferToRelayChain,
 	(
 		TakeWeightCredit,
-		AllowTopLevelPaidExecutionFrom<Everything>,
-		// Parent and its exec plurality get free execution
-		AllowUnpaidExecutionFrom<ParentOrParentsExecutivePlurality>,
 		// Expected responses are OK.
 		AllowKnownQueryResponses<PolkadotXcm>,
-		// Subscriptions for version tracking are OK.
-		AllowSubscriptionsFrom<ParentOrSiblings>,
+		// Allow XCMs with some computed origins to pass through.
+		WithComputedOrigin<
+			(
+				// If the message is one that immediately attemps to pay for execution, then allow it.
+				AllowTopLevelPaidExecutionFrom<Everything>,
+				// Parent and its executive plurality (i.e. governance) gets free execution.
+				AllowExplicitUnpaidExecutionFrom<ParentOrParentsExecutivePlurality>,
```

Comment:
Will this allow the Fellowship (IIRC `Technical` plurality) to execute? (@muharem)

Reply:
No, it specifically checks whether the body ID is `Executive`. I can relax this requirement so that _any_ plurality coming from the parent is allowed free execution.

Reply:
We have two uses (so far) of Gov2 origins being used on Statemine:

- XCMP Queue [ControllerOrigin](https://github.com/paritytech/cumulus/blob/dffcf83/parachains/runtimes/assets/statemine/src/lib.rs#L469)
- [CollatorSelectionUpdateOrigin](https://github.com/paritytech/cumulus/blob/dffcf83/parachains/runtimes/assets/statemine/src/lib.rs#L522)

I think it should handle messages from those origins without fee.

Reply:
Okay, changed it so that statemint/e and westmint accepts XCMs from any relay chain plurality.